internal/workflow/replicator.go

```diff
@@ -82,6 +89,11 @@ func (w *Replicator) execute(ctx workflow.Context, request *ReplicatorRequest) e
 			return xerrors.Errorf("failed to read config: %w", err)
 		}
 
+		if !request.UpdateWatermark {
```

Comment:
This doesn't make sense, why would we override the request.UpdateWatermark?

Reply:
Yes. You are right. I just tried and found this is not correct. I will change this part